bundles/framework/announcements/view/AnnouncementsHandler.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { StateHandler, controllerMixin, Messaging } from 'oskari-ui/util';
+import { Message } from 'oskari-ui';
+import { ANNOUNCEMENTS_LOCALSTORAGE } from './Constants';
+
+// Handler for announcements. Handles state and service calls.
+
+const getMessage = (key, args) => <Message messageKey={key} messageArgs={args} bundleKey='announcements' />;
+
+class ViewHandler extends StateHandler {
+    constructor () {
+        super();
+        this.fetchAnnouncements();
+        this.state = {
+            modals: [],
+            announcements: [],
+            checked: false
+        };
+    }
+
+    fetchAnnouncements () {
+        jQuery.ajax({
+            type: 'GET',
+            dataType: 'json',
+            url: Oskari.urls.getRoute('Announcements'),
+            success: (pResp) => {
+                this.updateState({
+                    announcements: pResp.data
+                });
+                this.updateModals();
+            },
+            error: function (jqXHR, textStatus) {
+                Messaging.error(getMessage('messages.getFailed'));
+            }
+        });
+    }
+
+    updateModals () {
+        var modals = [];
+        this.state.announcements.forEach((announcement) => {
+            if (announcement.active && this.showModal(announcement.id)) {
+                // if announcement is active, then show pop-up of the content
+                modals.push(announcement);
+            }
+        });
+        this.updateState({
+            modals: modals
+        });
+    }
+
+    setAnnouncementAsSeen (index, checked, id) {
+        if (checked) {
+            this.addToLocalStorageArray(ANNOUNCEMENTS_LOCALSTORAGE, id);
+            this.updateState({
+                checked: false
+            });
+        }
+        const newList = [...this.state.modals];
+        newList.splice(index, 1);
+        this.updateState({
+            modals: newList
+        });
+    }
+
+    showModal (id) {
```

Comment:
I would expect a modal to appear on the screen when calling a function with name `showModal()`. Rename to something like `isAnnouncementMarkedWithDontShow()`. You could make it better by checking if the announcement has active: true. Then it would be obvious why some are shown and others are not instead of the logic spread between functions. Then this could be called something like `isAnnouncementShown()`.